crates/polars-core/src/series/implementations/null.rs

```diff
@@ -280,6 +280,11 @@ impl SeriesTrait for NullChunked {
             // We still allow a length of `1` because it could be `lit(true)`.
             polars_ensure!(filter.len() <= 1, ShapeMismatch: "filter's length: {} differs from that of the series: 0", filter.len());
             0
+        } else if filter.len() == 1 {
+            return match filter.get(0) {
+                None | Some(true) => Ok(self.clone().into_series()),
```

Comment:
I believe that `None` should be in the `false` branch, not behave like `true`.

Reply:
Yeap, good catch.